dlt/pipeline/configuration.py

```diff
@@ -18,6 +19,8 @@ class PipelineConfiguration(BaseConfiguration):
     staging_name: Optional[str] = None
     loader_file_format: Optional[TLoaderFileFormat] = None
     dataset_name: Optional[str] = None
+    dataset_name_layout: Optional[str] = dlt.config.value
```

Comment:
it is OK to use it here but not necessary for correctly typed fields.
`dlt.config.value` and `dlt.secrets.value` are mostly used on `dlt.source` and `dlt.resource` function signatures to mark arguments that are required to be present and (when user is not adding types to arguments) to indicate which arguments contain secrets.

internally we convert function signatures into `configspec` (dataclasses and use those markers to correctly type them, mark as optional etc.)

in other words: as long as you create SPECs (not signatures) you do not need to use the marker above